File: Source/Managed/Editor/Windows/ImGuiDemoWindow.cs
using March.Core.Interop;

namespace March.Editor.Windows
{
    [EditorWindowMenu("Window/Dear ImGui Demo")]
    internal partial class ImGuiDemoWindow : EditorWindow
    {
        public ImGuiDemoWindow() : base(New(), "Dear ImGui Demo") { }

        [NativeMethod]
        private static partial nint New();
    }
}

File: Source/Native/EditorApp/Private/ImGuiDemoWindow.cpp
#include "pch.h"
#include "ImGuiDemoWindow.h"
#include "imgui.h"

namespace march
{
    bool ImGuiDemoWindow::Begin()
    {
        return true;
    }

    void ImGuiDemoWindow::End()
    {

    }

    void ImGuiDemoWindow::OnDraw()
    {
        ImGui::ShowDemoWindow(&m_IsOpen);
    }
}
